eng/tools/typespec-validation/src/rules/compile.ts

```diff
@@ -14,21 +17,127 @@ export class CompileRule implements Rule {
 
     if (await host.checkFileExists(path.join(folder, "main.tsp"))) {
       let [err, stdout, stderr] = await host.runCmd(
-        `npm exec --no -- tsp compile --warn-as-error ${folder}`,
+        `npm exec --no -- tsp compile --list-files --warn-as-error ${folder}`,
       );
+
+      stdOutput += stdout;
+
+      // Rule output is easier to read if "tsp compile" stderr is redirected to stdOutput
+      stdOutput += stderr;
+
       if (
         stdout.toLowerCase().includes("no emitter was configured") ||
         stdout.toLowerCase().includes("no output was generated")
       ) {
         success = false;
         errorOutput += "No emitter was configured and/or no output was generated.";
       }
-      if (err) {
-        success = false;
-        errorOutput += err.message;
+
+      if (success) {
+        if (!err) {
+          // Check for *extra* typespec-generated swagger files under the output folder, which
+          // indicates a mismatch between TypeSpec and swaggers.
+
+          // Example 'stdout':
+          //
+          // TypeSpec compiler v0.67.2
+          //
+          // ../resource-manager/Microsoft.Contoso/stable/2021-11-01/contoso.json
+          // ../resource-manager/Microsoft.Contoso/stable/2021-11-01/examples/Operations_List.json
+          //
+          // Compilation completed successfully.
+
+          // Remove ANSI color codes, handle windows and linux line endings
+          const lines = stripAnsi(stdout).split(/\r?\n/);
+
+          // TODO: Use helpers in /.github once they support platform-specific paths
+          const outputSwaggers = lines
+            // Remove leading and trailing whitespace
+            .map((l) => l.trim())
+            // Skip blank lines
+            .filter((l) => l !== "")
+            // Skip first and last lines (version and success messages)
+            .slice(1, -1)
```

Comment:
is this one necessary if you already try to filter by `.json` later, feel like this is something more likely to break the logic